databricks/src/rf_test/train.py

```diff
@@ -0,0 +1,151 @@
+
+    df = load_data(fpath, compute_type=mode)
+
+    X = df.drop(["ArrDelayBinary"], axis=1)
+    y = df["ArrDelayBinary"].astype('int32')
+
+    if mode == "GPU":
+        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)
+        mod = RandomForestClassifier(max_depth=max_depth, max_features=max_features, n_estimators=n_estimators)
+        acc_scorer = accuracy_score
+    elif mode == "CPU":
+        X_train, X_test, y_train, y_test = sk_split(X, y, test_size=0.2)
+        mod = sk_RF(max_depth=max_depth, max_features=max_features, n_estimators=n_estimators)
+        acc_scorer = sk_acc
+    else:
+        raise RuntimeError("Unknown option. Choose between [CPU, GPU], case sensitive.")
+
+    mod.fit(X_train, y_train)
+    preds = mod.predict(X_test)
+    acc = acc_scorer(y_test, preds)
+
+    mlparams = {"max_depth": str(max_depth),
+                "max_features": str(max_features),
+                "n_estimators": str(n_estimators),
+                "mode": str(mode)}
+    mlflow.log_params(mlparams)
+
+    mlmetrics = {"accuracy": acc}
+    mlflow.log_metrics(mlmetrics)
+
+    if (not hyperopt):
+        return mod
+
+    return {'loss': acc, 'status': STATUS_OK}
+
+
+def train(params, fpath, mode='GPU', hyperopt=False):
+    """
+    Proxy function used to call _train
+    :param params: hyperparameters. Its structure is consistent with how search space is defined. See below.
+    :param fpath: Path or URL for the training data used with the model.
+    :param mode: Hardware backend to use for training [CPU|GPU]
+    :param hyperopt: Use hyperopt for hyperparameter search during training.
+    :return: dict with fields 'loss' (scalar loss) and 'status' (success/failure status of run)
+    """
+    with mlflow.start_run(nested=True):
+        return _train(params, fpath, mode, hyperopt)
+
+
+if (__name__ in ("__main__",)):
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--algo', default='tpe', choices=['tpe'], type=str)
+    parser.add_argument('--conda-env', required=True, type=str)
+    parser.add_argument('--fpath', required=True, type=str)
+    parser.add_argument('--mode', default='GPU', choices=['GPU', 'CPU'], type=str)
+    args = parser.parse_args()
+
+    search_space = [
+        hp.uniform('max_depth', 5, 20),
+        hp.uniform('max_features', 0., 1.0),
+        hp.uniform('n_estimators', 150, 1000)
+    ]
+
+    trials = Trials()
+    algorithm = tpe.suggest if args.algo == 'tpe' else None
+    fn = partial(train, fpath=args.fpath, mode=args.mode, hyperopt=True)
+    experid = 0
+
+    with mlflow.start_run():
+        mlflow.set_tag("mlflow.runName", "RAPIDS-Hyperopt-Databricks")
+        argmin = fmin(fn=fn,
+                      space=search_space,
+                      algo=algorithm,
+                      max_evals=2,
+                      trials=trials)
+
+        print("===========")
+        fn = partial(train, fpath=args.fpath, mode=args.mode, hyperopt=False)
```

Comment:
I think everything is very clean and easy to understand up until this point, then it's sort of unclear why we're rummaging around with the conda environment. Can we separate out the conda writing (probably optional?) from core mlflow stuff?